src/lib/model/stateMachine/states/uploadPeerJWS.ts

```diff
@@ -93,7 +98,17 @@ export namespace UploadPeerJWS {
             }),
           onDone: {
             target: 'idle',
-            actions: [assign({ peerJWS: (_context, event) => event.data.updatedPeerJWS })],
+            actions: [
+              assign({
+                peerJWS: (_context, event) => {
+                  if (!event?.data) {
+                    opts.logger.warn('No event.data in uploadingPeerJWS.onDone');
+                    return [];
+                  }
+                  return event.data.updatedPeerJWS;
+                },
+              }),
+            ],
```

Comment:
the idea of this fix is preventing error in case if `event.data` is _null_ or _undefined_. if you sure that `event.data` is always an object, I can revert this change.   But based on logs from the issue (https://infitx-technologies.atlassian.net/browse/CSI-416), we might have such case